runtime/onert/api/nnfw/src/nnfw_api_internal.cc

```diff
@@ -1833,3 +1839,109 @@ NNFW_STATUS nnfw_session::quantize()
 
   return NNFW_STATUS_NO_ERROR;
 }
+
+NNFW_STATUS nnfw_session::set_compiled_model_path(const char *path)
+{
+  try
+  {
+    if (!isStateModelLoaded())
+    {
+      std::cerr << "invalid state" << std::endl;
+      return NNFW_STATUS_INVALID_STATE;
+    }
+
+    assert(_codegen_manager != nullptr);
+    _codegen_manager->exportModelPath(std::string(path));
+  }
+  catch (const std::exception &e)
+  {
+    std::cerr << "Error during nnfw_session::set_compiled_model_path : " << e.what() << std::endl;
+    return NNFW_STATUS_ERROR;
+  }
+
+  return NNFW_STATUS_NO_ERROR;
+}
+
+NNFW_STATUS nnfw_session::compile(const char *target, NNFW_COMPILE_PREF pref)
+{
+  try
+  {
+    if (!isStateModelLoaded())
+    {
+      std::cerr << "invalid state" << std::endl;
+      return NNFW_STATUS_INVALID_STATE;
+    }
+
+    std::string target_str{target};
+    if (target_str.empty() || target_str.substr(target_str.size() - 4) != "-gen")
+    {
+      std::cerr << "invalid target" << std::endl;
+      return NNFW_STATUS_ERROR;
+    }
+
+    onert::odc::CodegenPreference codegen_pref;
+    switch (pref)
+    {
+      case NNFW_COMPILE_PREF_DEFAULT:
+        codegen_pref = onert::odc::CodegenPreference::CODEGEN_PREF_DEFAULT;
+        break;
+      case NNFW_COMPILE_PREF_PERFORMANCE_FIRST:
+        codegen_pref = onert::odc::CodegenPreference::CODEGEN_PREF_PERFORMANCE_FIRST;
+        break;
+      case NNFW_COMPILE_PREF_MEMORY_FIRST:
+        codegen_pref = onert::odc::CodegenPreference::CODEGEN_PREF_MEMORY_FIRST;
+        break;
+      case NNFW_COMPILE_PREF_COMPILE_TIME_FIRST:
+        codegen_pref = onert::odc::CodegenPreference::CODEGEN_PREF_COMPILE_TIME_FIRST;
+        break;
+      default:
+        std::cerr << "invalid preference" << std::endl;
+        return NNFW_STATUS_ERROR;
+    }
+
+    assert(_codegen_manager != nullptr);
+    auto export_model_path = _codegen_manager->exportModelPath();
+    // If the export_model_path is not set, it generates a compiled model path
+    // automatically.
+    if (export_model_path.empty())
```

Comment:
These code(Set the proper model path with some vars) can be in CodegenManger.

Reply:
@YongseopKim I agree to implement this function in CodegenManager. I will update it in another PR. :)